### PR TITLE
Describe version attribute in the grammar

### PR DESCRIPTION
**Pull request by sfricke-samsung**
CC: @johnkslang @dneto0

### CODE REVIEW

File: chapters/creating_extension.md [hunk 1]
```diff
@@ -117,19 +117,52 @@ For example, there is a list of `BuiltIn` enums, and its `PointSize` entry is:
...
+- The `version` attribute corresponds to the
+  "[missing before](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#Unified)"
+  annotation in the SPIR-V specification.
+  If present, the `version` attribute indicates the first SPIR-V version in which the feature exists
+  as part of the core specification, without the need to use an `OpExtension` instruction.
+- The `extension` attribute, if present, indicates the name of the extension to use
+  with the `OpExtension` instruction when the feature is not yet part of the core SPIR-V specification.
+  The extension attribute takes the form of a list, in case a feature can be exposed through multiple
+  extensions.
+- If absent, the enumerant is available in all versions of SPIR-V
```
Comment (johnkslang) [inline chapters/creating_extension.md:133]: If this is describing the spec, that's what absent means.

If describing the grammar, possibly refer to the README for SPIRV-Headers (https://github.com/KhronosGroup/SPIRV-Headers), which includes this:

### Reserving tokens in the JSON grammar

Care should be taken to follow existing precedent in populating the details of reserved tokens.
This includes:
- pointing to what extension has more information, when possible
- keeping enumerants in numeric order
- when there are aliases, listing the preferred spelling first
- adding the statement `"version" : "None"`

The only requirement would be to be consistent with this.

File: chapters/creating_extension.md [hunk 2]
```diff
@@ -117,19 +117,52 @@ For example, there is a list of `BuiltIn` enums, and its `PointSize` entry is:
...
+
 
-For example, this is the entry for the `SubgroupBallotKHR` capability:
+See [2.22 Unified Specification](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#Unified)
+in the SPIR-V specification for more information.
+
+As a design principle, use the `extension` attribute on the capabilities introduced by the extension.
```
Comment (johnkslang) [inline chapters/creating_extension.md:139]: Not sure what design principles we're discussing.  Can't we just say

> Capabilities must declare what extensions introduce them.

File: chapters/creating_extension.md [hunk 3]
```diff
@@ -117,19 +117,52 @@ For example, there is a list of `BuiltIn` enums, and its `PointSize` entry is:
...
+Other tokens introduced by the extension are in turn guarded by the corresponding capability.
```
Comment (johnkslang) [inline chapters/creating_extension.md:140]: Maybe the possibility of relying on transitivity, which I don't think is a requirement but can work, is why a principle was discussed.  But, that's here, not up above.

File: chapters/creating_extension.md (hunk 2 above)
Comment (johnkslang) [reply chapters/creating_extension.md:139]: That is, it's actually the `"version" : "None"` that is required, and the capability declaring an extension that is optional.

File: chapters/creating_extension.md [hunk 4]
```diff
@@ -117,19 +117,52 @@ For example, there is a list of `BuiltIn` enums, and its `PointSize` entry is:
   },
 ```
 
-An enum entry can also have an `extensions` attribute to list the SPIR-V extensions that enable the feature. As a design principle, we only use these on the capabilities introduced by the extension. Other tokens introduced by the extension are in turn guarded by the corresponding capability.
+Enum entries within a list should be ordered by their value. The spec generator writes table entries in the same order as JSON file order.
```
Comment (johnkslang) [inline chapters/creating_extension.md:120]: Entries should be ordered by their enumerant value to make it easier to merge multiple versions together (from Khronos, public, and multiple PRs).

Delete the part of the spec generator, it's not true.

**Comment by dneto0** [comment]
I couldn't update this PR, so please see the update in #6

**Comment by sfricke-samsung** [comment]
Sounds good, this was really just to move the original internal discussion out here to get merged in... will close this and work off the new fix.

**Pull request closed without merging**